### PR TITLE
ELPP-2452 Added cache headers to recommendations

### PR DESCRIPTION
**Pull request by stephenwf**

### CODE REVIEW

**Comment by stephenwf** [comment]
Hi @thewilkybarkid  we don't currently have a web test foundation set up for recommendations, would you like me to set that up for this? If so is there more test cases that you think should be covered?

**Comment by giorgiosironi** [comment]
@stephenwf we will also have to add `/ping` soon if it's not already there, with "don't cache anything" headers

File: src/App/Kernel.php [hunk 1]
```diff
@@ -482,18 +482,27 @@ public function validate(Request $request, Response $response)
...
                 'response' => $response,
             ]);
         }
+
+        return $response;
     }
 
     public function cache(Request $request, Response $response)
     {
+        $response->setMaxAge($this->app['config']['ttl']);
```
Comment (giorgiosironi) [inline src/App/Kernel.php]: is there a way to override this for particular responses? (if needed in the future)

File: src/App/Kernel.php [hunk 2]
```diff
@@ -430,7 +430,7 @@ public function handleException(Throwable $e): Response
             return new JsonResponse(array_filter([
                 'error' => $e->getMessage(),
                 'trace' => $this->app['config']['debug'] ? $e->getTraceAsString() : null,
-            ]), $e->getCode());
+            ]), $e->getCode() ? $e->getCode() : 500);
```
Comment (thewilkybarkid) [inline src/App/Kernel.php]: `$e->getCode() ?? Response::HTTP_INTERNAL_SERVER_ERROR`?

**Comment by thewilkybarkid** [comment]
We'll need a test for `/ping`.

File: src/App/Kernel.php (hunk 1 above)
Comment (giorgiosironi) [inline src/App/Kernel.php]: /ping has been added that does not use caching, make sure to merge `develop` in this branch to solve possible conflicts

**Comment by giorgiosironi** [comment]
Now that web test cases seem well supported, I agree; there is a smoke test but it doesn't check caching headers.

File: tests/src/Web/CacheHeadersTest.php [hunk 3]
```diff
@@ -35,6 +35,16 @@ public function testETag()
         $this->assertEquals('Accept', $response->headers->get('Vary'));
     }
 
+    public function testPing()
+    {
+        $this->newClient();
+        $this->jsonRequest('GET', '/ping');
+        $response = $this->getResponse();
+        $this->assertEquals(200, $response->getStatusCode());
+        $this->assertEquals('max-age=300, must-revalidate, no-cache, no-store, public, stale-if-error=86400, stale-while-revalidate=300', $response->headers->get('Cache-Control'));
```
Comment (thewilkybarkid) [inline tests/src/Web/CacheHeadersTest.php]: This needs to be the same as https://github.com/elifesciences/search/blob/develop/tests/src/Search/Web/PingTest.php#L25-L27

Comment (stephenwf) [inline tests/src/Web/CacheHeadersTest.php]: Okay I've added in a `PrivateResponse` class that bypasses the cache method completely.

**Pull request merged**